src/Flexdis86/Assembler.hs

```diff
@@ -253,29 +255,55 @@ matchOperandType oso ops =
     (QWordReg (Reg64 rno), OpType (Opcode_reg rcode) VSize) -> rno `mod` 8 == rcode
     _ -> False
 
+
+data AssembledInstruction a =
```

Comment:
@travitch said:

> my comment on that is that it would be worth a comment around there noting that the order of the fields in AssembledInstruction and AssembledModRmSibDisp must absolutely not change in any way for that to be correct